src/mechanics/integrationtypes/IntegrationTypeTensorProduct.cpp

```diff
@@ -0,0 +1,259 @@
+#ifdef ENABLE_VISUALIZE
+#include "visualize/VisualizeEnum.h"
+#endif // ENABLE_VISUALIZE
+
+#include "mechanics/integrationtypes/IntegrationTypeTensorProduct.h"
+#include "mechanics/elements/ElementShapeFunctions.h"
+#include "math/Legendre.h"
+#include <iostream>
+
+// constructor
+template<int TDim>
+NuTo::IntegrationTypeTensorProduct<TDim>::IntegrationTypeTensorProduct(size_t nIps,
```

Comment:
Make this an `int` rather than an unsigned type, because the implicit conversion will cause weird bugs ("no unsigned in interfaces").